### PR TITLE
Create dependency between custom record type and its fields

### PR DESCRIPTION
**Pull request by dantal4**
Force a custom record type and its fields to be deployed together in the same change group.

---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- Create dependency between custom record type and its fields

---
_User Notifications_: 
None

### CODE REVIEW

**Comment by dantal4** [comment]
@netama since you're familiar with this mechanism, I will appreciate your feedback here 🙏

File: packages/netsuite-adapter/src/dependency_changer.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ({ change }) => getChangeData(change).elemID.getFullName()
+  )
+  return changesWithKeys.map(({ key, change }) => {
+    const parentDependency = isFieldChange(change)
+      ? typeChanges[getChangeData(change).parent.elemID.getFullName()]
+      : undefined
+    return parentDependency
+      ? dependencyChange('add', parentDependency.key, key)
+      : undefined
+  }).filter(values.isDefined)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/dependency_changer.ts:36]: I find this styling more readable, but you can ignore it if you prefer yours.

```suggestion
  const parentToFieldChanges = _(changesWithKeys)
    .filter(({ change }) => isFieldChange(change))
    .groupBy(({ key, change }) => getChangeData(change).parent.elemID.getFullName())
  Object.entries(parentToFieldChanges)
    .flatMap(([parentId, fieldChanges]) => {
      const parentDependency = typeChanges[parentId])
      return parentDependency
        ? fieldChanges.map(({ key }) => dependencyChange('add', parentDependency.key, key)
        : []
  })
```

**Review by omrilit — COMMENTED** [review]
I had only a single styling nit here and the code looks great.
My only question is why you decided to do it in the adapter level, don't you think that we should do it in the core level?

**Comment by dantal4** [comment]
@omrilit I think that this is the proper way to do it, because this way we don't care how the core split groups - we just make sure that a type and its fields are not split.

**Comment by omrilit** [comment]
> @omrilit I think that this is the proper way to do it, because this way we don't care how the core split groups - we just make sure that a type and its fields are not split.

I agree that it'll work but I think that this should be the general behavior in core also for other adapters.
Would love to hear @netama & @adamro opinion here as well

**Review by adamro — COMMENTED** [review]
I tend to agree with Daniel's approach here regarding core and adapter. It seems like an adapter decision that we don't want in all adapters. We already have this pipe-line for the adapter to add these dependencyChangers and it seems a legit use-case.

**Comment by dantal4** [comment]
@adamro @netama  Maybe the default behavior in the core should be a two-direction dependency between type&fields, and the adapters can remove that dependency?
How does it work in other adapters?

**Pull request closed without merging**